datajoint/schemas.py

```diff
@@ -273,43 +276,46 @@ def drop(self, force=False):
         """
         Drop the associated schema if it exists
         """
-        self._assert_activation()
         if not self.exists:
-            logger.info("Schema named `{database}` does not exist. Doing nothing.".format(database=self.database))
+            logger.info("Schema named `{database}` does not exist. Doing nothing.".format(
+                database=self.database))
         elif (not config['safemode'] or
               force or
               user_choice("Proceed to delete entire schema `%s`?" % self.database, default='no') == 'yes'):
             logger.info("Dropping `{database}`.".format(database=self.database))
             try:
                 self.connection.query("DROP DATABASE `{database}`".format(database=self.database))
                 logger.info("Schema `{database}` was dropped successfully.".format(database=self.database))
-            except pymysql.OperationalError:
-                raise DataJointError("An attempt to drop schema `{database}` "
-                                     "has failed. Check permissions.".format(database=self.database))
+            except AccessError:
+                raise AccessError(
+                    "An attempt to drop schema `{database}` "
+                    "has failed. Check permissions.".format(database=self.database))
 
     @property
     def exists(self):
         """
         :return: true if the associated schema exists on the server
         """
-        self._assert_activation()
-        cur = self.connection.query("SHOW DATABASES LIKE '{database}'".format(database=self.database))
-        return cur.rowcount > 0
+        if self.database is None:
+            raise DataJointError("Schema must be activated first.")
+        return self.database is not None and (
+            self.connection.query("SHOW DATABASES LIKE '{database}'".format(
```

Comment:
We should avoid querying using syntax unique to MySQL.

```suggestion
            self.connection.query("SELECT schema_name FROM information_schema.schemata WHERE schema_name LIKE '{database}'".format(
```

Reply:
perhaps here instead of `LIKE` we should just use `=`

Reply:
Yea wasn't sure if I missed something on first pass but looked like there were no wildcard operators used so should be good.